common/src/main/java/com/wynntils/features/chat/InputTranscriptionFeature.java

```diff
@@ -118,22 +120,20 @@ public void onChatScreenKeyTyped(ChatScreenKeyTypedEvent event) {
         }
     }
 
-    @SubscribeEvent
-    public void onChatSend(ChatSentEvent event) {
-        if (event.getMessage().isBlank()) return;
-
-        String message = event.getMessage();
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onChatScreenSend(ChatScreenSendEvent event) {
+        // Parses brackets and does the transcription, then sends the new message.
+        if (event.getInput().isBlank()) return;
 
-        if (Models.WynnAlphabet.containsBrackets(message)) {
-            String updatedMessage = Models.WynnAlphabet.transcribeBracketedText(message);
+        String message = event.getInput();
+        if (!Models.WynnAlphabet.containsBrackets(message)) return;
 
-            updatedMessage = updatedMessage.substring(0, Math.min(updatedMessage.length(), MAX_CHAT_LENGTH));
+        String updatedMessage = Models.WynnAlphabet.transcribeBracketedText(message);
+        updatedMessage = updatedMessage.substring(0, Math.min(updatedMessage.length(), MAX_CHAT_LENGTH));
+        if (updatedMessage.equals(message)) return; // No changes, we can pretend nothing happened.
 
-            if (!updatedMessage.equals(message)) {
-                event.setCanceled(true);
-                McUtils.sendChat(updatedMessage);
-            }
-        }
+        event.setCanceled(true);
+        Services.ChatTab.sendChat(updatedMessage);
```

Comment:
So this will add a prefix, if present? I still think this interdependency is convoluted.

Reply:
Yes. It would be a bit strange if suddenly the chat tabs were not respected just because the user decided to type in Wynnic. I am not sure of a way to make this less interdependent, hooking into the native chat-sending method isn't the best either since we can't control if the prefix is added or not and it will always be added